### PR TITLE
Replace traditional for loop and clean constants

### PR DESCRIPTION
**Pull request by alanna-zhou**
#177 I've addressed the last traditional for loop that could be replaced with a forEach

Constants were also all cleaned up and put in central `Constants.js` file.

### CODE REVIEW

File: src/routers/v1/PlacesAutocompleteRouter.js [hunk 1]
```diff
@@ -33,12 +30,12 @@ class PlacesAutocompleteRouter extends ApplicationRouter<string> {
...
     const options = {
       method: 'GET',
-      url: 'https://maps.googleapis.com/maps/api/place/autocomplete/json',
+      url: Constants.GOOGLE_AUTOCOMPLETE_URL,
       qs: {
         input: query,
         key: process.env.PLACES_KEY,
         location: '42.4440,-76.5019',
-        radius: 24140,
+        radius: Constants.RADIUS,
```
Comment (meganle) [inline src/routers/v1/PlacesAutocompleteRouter.js]: Can we make this more descriptive like `Constants.AUTOCOMPLETE_RADIUS` or something

File: src/utils/Constants.js [hunk 2]
```diff
@@ -1,11 +1,82 @@
...
+const DEG_MIN_PRECISION = 1;
+const DEG_NEARBY_PRECISION = 4; // 4 degrees of precision is about 11 meters, stop nearby
+const DEG_WALK_PRECISION = 3; // 3 degrees of precision is about 111 meters, stop walkable
+
+/* String & URL constants */
+const CURRENT_LOCATION = 'Current Location';
+const GOOGLE_AUTOCOMPLETE_URL = 'https://maps.googleapis.com/maps/api/place/autocomplete/json';
+const GOOGLE_PLACES_URL = 'https://maps.googleapis.com/maps/api/place/details/json';
+const TOKEN_URL = 'https://gateway.api.cloud.wso2.com:443/token';
+const POSTMAN_TOKEN = '42201611-965d-4832-a4c5-060ad3ff3b83';
```
Comment (meganle) [inline src/utils/Constants.js]: is this necessary?

File: src/appdev/ApplicationAPI.js [hunk 3]
```diff
@@ -39,9 +39,9 @@ class ApplicationAPI {
     AppDevUtilities.tryCheckAppDevURL(this.getPath());
     const middleware = this.middleware();
     const routerGroups = this.routerGroups();
-    for (let i = 0; i < middleware.length; i++) {
-      this.express.use(middleware[i]);
-    }
+    middleware.forEach((m) => {
```
Comment (young-k) [inline src/appdev/ApplicationAPI.js]: Is `(m)` preferred over `m`? Can you check a style guide and see if that's preferred?

Comment (kq-li) [reply src/appdev/ApplicationAPI.js]: [airbnb](https://github.com/airbnb/javascript) says always include parens

File: src/utils/AllStopUtils.js [hunk 4]
```diff
@@ -71,7 +62,7 @@ async function fetchAllStops() {
 
 function fetchPrecisionMaps() {
   const maps = {};
-  maps[DEG_EQ_PRECISION] = getPrecisionMap(DEG_EQ_PRECISION);
+  maps[Constants.DEG_EQ_PRECISION] = getPrecisionMap(Constants.DEG_EQ_PRECISION);
```
Comment (kq-li) [inline src/utils/AllStopUtils.js]: this function is an easy target for refactoring for code clarity and quality

File: src/appdev/ApplicationAPI.js (hunk 3 above)
Comment (alanna-zhou) [reply src/appdev/ApplicationAPI.js]: ya i get an error without the parens

**Comment by young-k** [comment]
@alanna-zhou, should we close this PR?

**Pull request closed without merging**